### DISCUSSION TITLE
Question about image automation force push

### DISCUSSION THREAD

**Discussion by yoan-myparcel** [fluxcd/flux2#3752] [Q&A] [answered]
Hi Flux community!

I have a question about the force-pushes that result from ImageUpdateAutomation settings. I am currently running flux version 0.35.0 and I am investigating any breaking changes in upgrading to newer versions. I saw in the release notes of version 0.37.0 that if a spec.git.push.branch is specified on a ImageUpdateAutomation, then a git force-push will happen. 

My understanding is that if the field is omitted then a force-push will not happen to the branch defined in `spec.git.checkout.ref.branch` (or the one inherited through the related GitRepository). 

I wanted to confirm if that really is the case and that my understanding of the matter is correct. 

Thank you for the help!

**Comment by makkes** [comment, marked as answer]
Yes, force-push is only performed if and only if you pass a non-empty `.spec.git.push.branch`: https://github.com/fluxcd/image-automation-controller/blob/0fb73e6996b3a8a8227b9365313f571c73bd6fc1/internal/controllers/imageupdateautomation_controller.go#L265

**Comment by yoan-myparcel** [comment]
Thank you for the answer @makkes !
